src/Bicep.Cli/Services/ValidationService.cs

```diff
@@ -1,33 +1,55 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.IO;
 using System.Text;
 using Bicep.Core.Emit;
 using Bicep.Core.Intermediate;
 using Bicep.Core.Semantics;
 using Bicep.Core.Syntax;
+using Microsoft.Extensions.Logging;
 using Microsoft.WindowsAzure.ResourceStack.Common.Collections;
 using Microsoft.WindowsAzure.ResourceStack.Common.Json;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
 namespace Bicep.Cli.Services
 {
-    public class ValidationService
+    public class Validation
     {
-        public static void Validate(ImmutableArray<TestSymbol> testDeclarations)
+        private InsensitiveDictionary<TemplateEvaluation> successfullEvaluations;
+
+        private InsensitiveDictionary<TemplateEvaluation> failedEvaluations;
+
+        private InsensitiveDictionary<TemplateEvaluation> skippedEvaluations;
+
+        public InsensitiveDictionary<TemplateEvaluation> SuccessfullEvaluations => successfullEvaluations;
+
+        public InsensitiveDictionary<TemplateEvaluation> FailedEvaluations => failedEvaluations;
```

Comment:
Instead of the following:
```csharp
private InsensitiveDictionary<TemplateEvaluation> failedEvaluations;
public InsensitiveDictionary<TemplateEvaluation> FailedEvaluations => failedEvaluations;
```
You should be able to do this, which is functionally equivalent (except for allowing the entire dictionary to be overwritten, which I don't think you need):
```csharp
public InsensitiveDictionary<TemplateEvaluation> FailedEvaluations { get; }
```